src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,65 @@
+import cn from 'classnames';
+import { useMemo } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+
+const currentCenturies = ['16', '17', '18', '19', '20'];
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query') || '';
+  const centuries = useMemo(
+    () => searchParams.getAll('centuries') || [],
+    [searchParams],
+  );
+
+  function handleQueryChange(e: React.ChangeEvent<HTMLInputElement>) {
+    const inputValue = e.target.value;
+    const params = new URLSearchParams(searchParams);
+
+    if (inputValue) {
+      params.set('query', inputValue);
+    } else {
+      params.delete('query');
+    }
+
+    setSearchParams(params);
+  }
+
+  function toggleNumbers(num: string) {
+    const newCenturies = centuries.includes(num)
+      ? centuries.filter(c => c !== num)
+      : [...centuries, num];
+
+    return newCenturies;
+  }
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
+        <SearchLink
+          params={{ sex: null }}
+          className={cn({ 'is-active': !sex })}
+        >
           All
-        </a>
-        <a className="" href="#/people?sex=m">
+        </SearchLink>
+
+        <SearchLink
+          params={{ sex: 'm' }}
+          className={cn({ 'is-active': sex === 'm' })}
```

Comment:
Consider will be better if you create a enum for 'm' and 'f' and use it everywhere